index.js

```diff
@@ -319,36 +317,48 @@ const captureWebsite = async (input, options) => {
 	}
 
 	if (screenshotOptions.fullPage) {
-		// Get the height of the rendered page
-		const bodyHandle = await page.$('body');
-		const bodyBoundingHeight = await bodyHandle.boundingBox();
-		await bodyHandle.dispose();
-
-		// Scroll one viewport at a time, pausing to let content load
-		const viewportHeight = viewportOptions.height;
-		let viewportIncrement = 0;
-		while (viewportIncrement + viewportHeight < bodyBoundingHeight) {
-			const navigationPromise = page.waitForNavigation({waitUntil: 'networkidle0'});
-			/* eslint-disable no-await-in-loop */
-			await page.evaluate(_viewportHeight => {
-				/* eslint-disable no-undef */
-				window.scrollBy(0, _viewportHeight);
-				/* eslint-enable no-undef */
-			}, viewportHeight);
-			await navigationPromise;
-			/* eslint-enable no-await-in-loop */
-			viewportIncrement += viewportHeight;
+		const idleCallbackToken = 'CAPTURE_WEBSITE_SCROLL_READY';
+		const autoScroll = async () => {
+			// Scroll and check if done
+			const isBottom = await page.evaluate(idleCallbackToken => {
+				window[idleCallbackToken] = false;
+				window.scrollBy(0, window.innerHeight);
+
+				// Wait for idle event loop to ensure client-side rendering is complete
+				window.requestIdleCallback(() => {
+					window[idleCallbackToken] = true;
+				}, {timeout: 100});
```

Comment:
In addition to checking `image.complete` we're now pausing between scrolls for the event loop to clear via `requestIdleCallback`.  This lets any client side rendering (e.g. react, vue, etc.) which may add additional images complete before continuing scroll and checking image completion.  In testing, this adds next to no additional processing time.  However, it does fix issues where the scroller thinks it's complete just to have the page grow from JS execution after-the-fact.